plugin/src/main/java/org/opensearch/ml/model/MLModelAutoReLoader.java

```diff
@@ -0,0 +1,352 @@
+                throw new RuntimeException(e);
+            }
+        });
+    }
+
+    /**
+     * auto reload all the models under the node id<br> the node must be a ml node<br>
+     *
+     * @param localNodeId node id
+     */
+    @VisibleForTesting
+    void autoReLoadModelByNodeId(String localNodeId) throws ExecutionException, InterruptedException {
+        StepListener<IndicesExistsResponse> indicesExistsResponseStepListener = new StepListener<>();
+        StepListener<SearchResponse> queryTaskStep = new StepListener<>();
+        StepListener<SearchResponse> getReTryTimesStep = new StepListener<>();
+        StepListener<IndexResponse> saveLatestReTryTimesStep = new StepListener<>();
+
+        indicesExistsResponseStepListener.whenComplete(indicesExistsResponse -> {
+            if (indicesExistsResponse.isExists()) {
+                queryTask(localNodeId, ActionListener.wrap(queryTaskStep::onResponse, queryTaskStep::onFailure));
+            }
+        }, indicesExistsResponseStepListener::onFailure);
+
+        isExistedIndex(ML_TASK_INDEX, indicesExistsResponseStepListener);
+
+        getReTryTimes(localNodeId, ActionListener.wrap(getReTryTimesStep::onResponse, getReTryTimesStep::onFailure));
+
+        Result result = new Result();
+        queryTaskStep.whenComplete(searchResponse -> {
+            SearchHit[] hits = searchResponse.getHits().getHits();
+            if (CollectionUtils.isEmpty(hits)) {
+                return;
+            }
+            result.setHits(hits);
+
+            getReTryTimesStep.whenComplete(getReTryTimesResponse -> {
+                // if getReTryTimesResponse is null,it means we get reTryTimes at the first time,and the index
+                // .plugins-ml-model-reload doesn't exist,so we should let reTryTimes be zero(init value)
+                // we don't do anything
+                // if getReTryTimesResponse is not null,it means we have saved the value of reTryTimes into the index
+                // .plugins-ml-model-reload,so we get the value of the field MODEL_LOAD_RETRY_TIMES_FIELD
+                if (getReTryTimesResponse != null) {
+                    Map<String, Object> sourceAsMap = getReTryTimesResponse.getHits().getHits()[0].getSourceAsMap();
+                    result.setReTryTimes((Integer) sourceAsMap.get(MODEL_LOAD_RETRY_TIMES_FIELD));
+                }
+
+                // According to the node id to get retry times, if more than the max retry times, don't need to retry
+                // that the number of unsuccessful reload has reached the maximum number of times, do not need to reload
+                if (result.getReTryTimes() > autoReLoadMaxReTryTimes) {
+                    log.info("have exceeded max retry times, always failure");
+                    return;
+                }
+
+                int reTryTimes = 0;
+                try (XContentParser parser = createXContentParserFromRegistry(xContentRegistry, result.getHits()[0].getSourceRef())) {
```

Comment:
From `result.getHits()[0]`, seems it will only reload the first model?

Reply:
This `retryTimes` is defined based on the dimension of `node`, not `node+model`.
In other words, there is only one value of the `retryTimes` for each ml node